api/validation/api.go

```diff
@@ -0,0 +1,272 @@
+package validation
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/cloudtrust/keycloak-bridge/internal/dto"
+	"github.com/cloudtrust/keycloak-bridge/internal/keycloakb"
+	kc "github.com/cloudtrust/keycloak-client"
+)
+
+// UserRepresentation struct
+type UserRepresentation struct {
+	UserID               *string `json:"userId,omitempty"`
+	Username             *string `json:"username,omitempty"`
+	Gender               *string `json:"gender,omitempty"`
+	FirstName            *string `json:"firstName,omitempty"`
+	LastName             *string `json:"lastName,omitempty"`
+	EmailAddress         *string `json:"emailAddress,omitempty"`
+	EmailAddressVerified *bool   `json:"emailAddressVerified,omitempty"`
+	PhoneNumber          *string `json:"phoneNumber,omitempty"`
+	PhoneNumberVerified  *bool   `json:"phoneNumberVerified,omitempty"`
+	BirthDate            *string `json:"birthDate,omitempty"`
+	BirthLocation        *string `json:"birthLocation,omitempty"`
+	IDDocumentType       *string `json:"idDocumentType,omitempty"`
+	IDDocumentNumber     *string `json:"idDocumentNumber,omitempty"`
+	IDDocumentExpiration *string `json:"idDocumentExpiration,omitempty"`
+}
+
+// CheckRepresentation struct
+type CheckRepresentation struct {
+	UserID    *string `json:"userId,omitempty"`
+	Operator  *string `json:"operator,omitempty"`
+	DateTime  *int64  `json:"datetime,omitempty"`
+	Status    *string `json:"status,omitempty"`
+	Type      *string `json:"type,omitempty"`
+	Nature    *string `json:"nature,omitempty"`
+	ProofData *[]byte `json:"proofData,omitempty"`
+	ProofType *string `json:"proofType,omitempty"`
+}
+
+// Parameter references
+const (
+	prmUserID                   = "user_id"
+	prmUserGender               = "user_gender"
+	prmUserFirstName            = "user_firstName"
+	prmUserLastName             = "user_lastName"
+	prmUserEmail                = "user_emailAddress"
+	prmUserPhoneNumber          = "user_phoneNumber"
+	prmUserBirthDate            = "user_birthDate"
+	prmUserBirthLocation        = "user_birthLocation"
+	prmUserIDDocumentType       = "user_idDocType"
+	prmUserIDDocumentNumber     = "user_idDocNumber"
+	prmUserIDDocumentExpiration = "user_idDocExpiration"
+
+	prmCheckOperator  = "check_operator"
+	prmCheckDatetime  = "check_datetime"
+	prmCheckStatus    = "check_status"
+	prmCheckType      = "check_type"
+	prmCheckNature    = "check_nature"
+	prmCheckProofType = "check_proof_type"
+
+	RegExpID            = `^[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$`
+	regExpNames         = `^([\wàáâäçèéêëìíîïñòóôöùúûüß]+([ '-][\wàáâäçèéêëìíîïñòóôöùúûüß]+)*){1,50}$`
```

Comment:
I also subscribe to it. However it would need to refactor it globally and according to our courent time constraint it is not reasonable to launch this task right now.